includes/payments/functions.php

```diff
@@ -1511,17 +1510,24 @@ function edd_get_payment_notes( $payment_id = 0, $search = '' ) {
  * Add a note to a payment
  *
  * @since 1.4
- * @param int $payment_id The payment ID to store a note for
- * @param string $note The note to store
- * @return int The new note ID
+ * @since 3.0 Updated to use the edd_notes custom table to store notes.
+ *
+ * @param int    $payment_id The payment ID to store a note for.
+ * @param string $note       The content of the note.
+ * @return int|false The new note ID, false otherwise.
  */
 function edd_insert_payment_note( $payment_id = 0, $note = '' ) {
-	if ( empty( $payment_id ) )
+	if ( empty( $payment_id ) ) {
 		return false;
+	}
 
 	do_action( 'edd_pre_insert_payment_note', $payment_id, $note );
 
-	$note_id = wp_insert_comment( wp_filter_comment( array(
+	/**
+	 * For backwards compatibility purposes, we need to pass the data to wp_filter_comment in the event that the note
+	 * data is filtered using the WordPress Core filters prior to be inserted into the database.
+	 */
+	$filtered_data = wp_filter_comment( array(
```

Comment:
Do we have a realistic case where this is actually needed?

Reply:
It's here purely for backwards compatibility on the off chance that it gets filtered.  Highly unlikely that it will ever be  needed but doesn't hurt to keep it there?